public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistoryScene.tsx

```diff
@@ -189,71 +181,72 @@ function getSceneQuery(datasource: DataSourceInformation) {
  * The query uses a runtime datasource that fetches the events from the history api.
  */
-export function getEventsScenesFlexItem(datasource: DataSourceInformation) {
+export function getEventsScenesFlexItem() {
   return new SceneFlexItem({
     minHeight: 300,
-    body: PanelBuilders.timeseries()
-      .setTitle('Alert Events')
-      .setDescription(
-        'Each alert event represents an alert instance that changed its state at a particular point in time. The history of the data is displayed over a period of time.'
-      )
-      .setData(getSceneQuery(datasource))
-      .setColor({ mode: 'continuous-BlPu' })
-      .setCustomFieldConfig('fillOpacity', 100)
-      .setCustomFieldConfig('drawStyle', GraphDrawStyle.Bars)
-      .setCustomFieldConfig('lineInterpolation', LineInterpolation.Linear)
-      .setCustomFieldConfig('lineWidth', 1)
-      .setCustomFieldConfig('barAlignment', 0)
-      .setCustomFieldConfig('spanNulls', false)
-      .setCustomFieldConfig('insertNulls', false)
-      .setCustomFieldConfig('showPoints', VisibilityMode.Auto)
-      .setCustomFieldConfig('pointSize', 5)
-      .setCustomFieldConfig('stacking', { mode: StackingMode.None, group: 'A' })
-      .setCustomFieldConfig('gradientMode', GraphGradientMode.Hue)
-      .setCustomFieldConfig('scaleDistribution', { type: ScaleDistribution.Linear })
-      .setOption('legend', { showLegend: false, displayMode: LegendDisplayMode.Hidden })
-      .setOption('tooltip', { mode: TooltipDisplayMode.Single })
+    body:
+      // eslint-disable-next-line
+      PanelBuilders.timeseries()
+        .setTitle('Alert Events')
+        .setDescription(
+          'Each alert event represents an alert instance that changed its state at a particular point in time. The history of the data is displayed over a period of time.'
+        )
+        .setData(getQueryRunnerForAlertHistoryDataSource())
+        .setColor({ mode: 'continuous-BlPu' })
+        .setCustomFieldConfig('fillOpacity', 100)
+        .setCustomFieldConfig('drawStyle', GraphDrawStyle.Bars)
+        .setCustomFieldConfig('lineInterpolation', LineInterpolation.Linear)
+        .setCustomFieldConfig('lineWidth', 1)
+        .setCustomFieldConfig('barAlignment', 0)
+        .setCustomFieldConfig('spanNulls', false)
+        .setCustomFieldConfig('insertNulls', false)
+        .setCustomFieldConfig('showPoints', VisibilityMode.Auto)
+        .setCustomFieldConfig('pointSize', 5)
+        .setCustomFieldConfig('stacking', { mode: StackingMode.None, group: 'A' })
+        .setCustomFieldConfig('gradientMode', GraphGradientMode.Hue)
+        .setCustomFieldConfig('scaleDistribution', { type: ScaleDistribution.Linear })
+        .setOption('legend', { showLegend: false, displayMode: LegendDisplayMode.Hidden })
+        .setOption('tooltip', { mode: TooltipDisplayMode.Single })
+        .setNoValue('No events found')
+        .build() as VizPanel<Options, FieldConfig>,
```

Comment:
Is the type casting necessary here?

Reply:
yes

Reply:
I think is not anymore (it comes for some changes I did before the final version)! good catch